server/src/main/java/org/elasticsearch/repositories/blobstore/FileRestoreContext.java

```diff
@@ -188,12 +188,12 @@ private void afterRestore(SnapshotFiles snapshotFiles, Store store, StoreFileMet
 
         /// now, go over and clean files that are in the store, but were not in the snapshot
         try {
+            store.deleteQuiet("restore");
             for (String storeFile : store.directory().listAll()) {
                 if (Store.isAutogenerated(storeFile) || snapshotFiles.containPhysicalIndexFile(storeFile)) {
                     continue; // skip write.lock, checksum files and files that exist in the snapshot
```

Comment:
```suggestion
                    continue; // skip write.lock and files that exist in the snapshot
```

"checksum files" are not a thing any more (since 5.0?); the javadoc on `Store#isAutogenerated` has the same cruft as this.